pkg/lib/machines.js

```diff
@@ -554,15 +555,57 @@
                     state(host, "connecting", null);
             }
 
+            function load_manifest_plugins(packaged_manifests) {
+                if (manifest_plugins[host]) {
+                    manifest_plugins[host].forEach(function (pl) {
+                        pl.close();
+                        $(pl).off();
+                    });
+                }
+                manifest_plugins[host] = [ ];
+
+                function update_manifests() {
+                    var all_manifests = { };
+                    $.extend(true, all_manifests, packaged_manifests);
+                    manifest_plugins[host].forEach(function (pl) {
+                        if (pl.manifests)
+                            $.extend(true, all_manifests, pl.manifests);
+                    });
+                    machines.overlay(host, { manifests: all_manifests });
+                }
+
+                // XXX - This loads and evals all plugins in one go.
+                // Instead, we might want to eval them one by one so
+                // that a crash in one plugin does not disable all
+                // subsequent plugins.
+
+                $.ajax({ url: url + "/*/manifestplugin.js", dataType: "text", cache: true})
```

Comment:
For example, are they subentries of other entries? In which case the task switch redesign should account for that.